ts/Stock/Indicators/MACD/MACDIndicator.ts

```diff
@@ -0,0 +1,535 @@
+             */
+            zones: [],
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line.
+                 *
+                 * @type  {Highcharts.ColorString}
+                 */
+                lineColor: void 0
+            }
+        },
+        /**
+         * The styles for macd line
+         */
+        macdLine: {
+            /**
+             * @sample stock/indicators/macd-zones
+             *         Zones in MACD
+             *
+             * @extends plotOptions.macd.zones
+             */
+            zones: [],
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line.
+                 *
+                 * @type  {Highcharts.ColorString}
+                 */
+                lineColor: void 0
+            }
+        },
+        /**
+         * @type {number|null}
+         */
+        threshold: 0,
+        groupPadding: 0.1,
+        pointPadding: 0.1,
+        crisp: false,
+        states: {
+            hover: {
+                halo: {
+                    size: 0
+                }
+            }
+        },
+        tooltip: {
+            pointFormat: '<span style="color:{point.color}">\u25CF</span> <b> {series.name}</b><br/>' +
+                'Value: {point.MACD}<br/>' +
+                'Signal: {point.signal}<br/>' +
+                'Histogram: {point.y}<br/>'
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        },
+        minPointLength: 0
+    } as MACDOptions);
+}
+
+interface MACDIndicator {
+    applyZones(): void;
+    crispCol: ColumnSeries['crispCol'];
+    currentLineZone: (string|null);
+    data: Array<MACDPoint>;
+    destroy(): void;
+    drawGraph(): void;
+    drawPoints: ColumnSeries['drawPoints'];
+    getColumnMetrics: ColumnSeries['getColumnMetrics'];
+    getValues<TLinkedSeries extends LineSeries>(
+        series: TLinkedSeries,
+        params: MACDParamsOptions
+    ): (IndicatorValuesObject<TLinkedSeries>|undefined);
+    getZonesGraphs(
+        props: Array<Array<string>>
+    ): Array<Array<string>>;
+    graphmacd: (Highcharts.SVGElement|undefined);
+    graphsignal: (Highcharts.SVGElement|undefined);
+    init(): void;
+    macdZones: Highcharts.MACDIndicatorZonesOptions;
+    nameComponents: Array<string>;
+    options: Highcharts.MACDIndicatorOptions;
+    parallelArrays: Array<string>;
+    pointArrayMap: Array<string>;
+    pointClass: typeof Highcharts.MACDIndicatorPoint;
+    points: Array<Highcharts.MACDIndicatorPoint>;
+    pointValKey: string;
+    requiredIndicators: Array<string>;
+    signalZones: Highcharts.MACDIndicatorZonesOptions;
+    toYData(point: Point): Array<number>;
+    translate(): void;
```

Comment:
Regular properties (not mentioned in extend call, excepted pointClass) and functions got into the class pattern.